crates/sui-indexer/src/handlers/event_handler.rs

```diff
@@ -47,13 +56,19 @@ impl EventHandler {
         loop {
             let event_page = fetch_event_page(self.rpc_client.clone(), next_cursor).await?;
             let event_count = event_page.data.len();
+            self.event_handler_metrics
+                .total_events_received
+                .inc_by(event_count as u64);
             commit_events(&mut pg_pool_conn, event_page.clone())?;
             commit_event_log(
                 &mut pg_pool_conn,
                 event_page.next_cursor.clone().map(|c| c.tx_seq),
                 event_page.next_cursor.clone().map(|c| c.event_seq),
             )?;
```

Comment:
There's a subtle bug here: say `event_page.next_cursor` is `None` because it gets to the latest page, the indexer might skip some events in the following cases:
1. Say `event_count = EVENT_PAGE_SIZE - delta` and the handler goes to sleep on line 73, there could be more than `EVENT_PAGE_SIZE + delta` events committed when the handler is asleep. But in the next loop on line 57, the handler is going to fetch the latest `EVENT_PAGE_SIZE` events and skip the delta;
2. The handler crashes and gets restarted, it will fetch only the latest transaction since `next_cursor` is `None`

We should not commit event log when `next_cursor` is `None`

Reply:
Ah interesting, I did not realize that event_page.next_cursor can be None if it's the latest page already.

> the handler is going to fetch the latest EVENT_PAGE_SIZE events and skip the delta;

actually I think b/c `next_cursor` is None, indexer will start fetching events from scratch per [pointer](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority.rs#L2164-L2168)

Ideally even though the latest page has been reached, we can still have a way to mark it as complete on the indexer end, but that is a secondary thing. Will add a fix here, thanks Chris!